src/switch_core_session.c

```diff
@@ -1459,6 +1459,78 @@ SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_events(sw
 	return x;
 }
 
+SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_media_events(switch_core_session_t *session)
+{
+	switch_status_t status = SWITCH_STATUS_FALSE;
+	switch_queue_t *queue_pri, *queue;
+	int x = 0;
+	void *pop;
+	
+	if (session->private_event_queue) {
+		switch_memory_pool_t *pool = switch_core_session_get_pool(session);
+		switch_queue_create(&queue_pri, SWITCH_EVENT_QUEUE_LEN, pool);
+		switch_queue_create(&queue, SWITCH_EVENT_QUEUE_LEN, pool);
+
+		while ((status = (switch_status_t) switch_queue_trypop(session->private_event_queue_pri, &pop)) == SWITCH_STATUS_SUCCESS) {
+			if (pop) {
+				switch_event_t *interrupt_event = NULL;
+				switch_event_t *command_event = (switch_event_t *) pop;
+				const char *app_name = switch_event_get_header(command_event, "execute-app-name");
+
+				if (!strcasecmp(app_name, "playback") || !strcasecmp(app_name, "speak") || !strcasecmp(app_name, "say")) {
```

Comment:
I wouldn't recommend to specify the command name inside switch core since it should be abstracted. The core should not have any idea about command names implemented by a module such as mod_commands or mod_call_control. I would recommend that this command or api names should be given as a parameter in this function